src/python/WMCore/Services/Dashboard/DashboardAPI.py

```diff
@@ -32,19 +32,26 @@ def __init__(self, monitorId=None, jobMonitorId=None, lookupUrl=None,
         self.defaultParams = {'sys_monitoring': 0, 'general_info': 0, 'job_monitoring': 0}
         self.server = server if server else DASHBOARDURL
         self.logger = logr if logr else logging.getLogger()
+        self.apmon = self._getApMonInstance()
 
-    def getApMonInstance(self):
+    def __enter__(self):
+        return self
```

Comment:
Not to say that you should change it, but the way around that is self.apmon = None in the init()